storage/namespace.go

```diff
@@ -522,7 +551,23 @@ func (n *dbNamespace) Flush(
 	return res
 }
 
-func (n *dbNamespace) NeedsFlush(blockStart time.Time) bool {
+func (n *dbNamespace) NeedsFlush(start time.Time, end time.Time) bool {
```

Comment:
It might be simpler and easier to reason about if the function only takes a single time as the parameter and let the caller iterate over the list of start times.